packages/client/src/clients/guide/client.ts

```diff
@@ -908,6 +951,21 @@ export class KnockGuideClient {
     if (this.store.state.location === href) return;
 
     this.knock.log(`[Guide] Handle Location change: ${href}`);
+
+    const newDebugParams = detectDebugParams();
+    const currentDebugParams = this.store.state.debug;
+
+    if (newDebugParams.forcedGuideKey !== currentDebugParams.forcedGuideKey) {
+      this.knock.log(`[Guide] Debug params changed`);
+
+      this.store.setState((state) => ({
+        ...state,
+        debug: newDebugParams,
+      }));
+
+      this.fetch();
+    }
+
```

Comment:
A couple things:
* `this.setLocation(href)` also updates the state, and ideally we call setState only once with the new debug params if necessary?
* We call `this.fetch()` whenever `forcedGuideKey` changes, but ideally we only call it when we go from non-debug mode to debug mode?

Not blocking / huge issues while we work on the feature, but something we want to handle before it goes out.